f5_openstack_agent/lbaasv2/drivers/bigip/exceptions.py

```diff
@@ -356,3 +364,78 @@ class VXLANDeleteException(F5AgentException):
 
 class BigIPNotLicensedForVcmp(F5AgentException):
     pass
+
+
+class F5MissingDependencies(F5AgentException):
+    default_msg = "%s cannot start due to missing dependency" % \
+        str(sys.argv[0])
+    message_format = "(%d) %s: %s [%s:%s]"
+    default_errno = errno.ENOSYS
+    default_project = 'neutron'
+    default_name = 'f5-oslbaasv2-agent'
+
+    def __init__(self, *args, **kargs):
+        self.__set_message(args, kargs)
+        super(F5MissingDependencies, self).__init__(self.message)
+        self.__logger()
+        self.__log_error()
+        self.__check_debug()
+
+    def __check_debug(self):
+        print("__check_debug")
+        debug = False
+        for item in sys.argv:
+            if 'f5-openstack-agent.ini' in item:
+                with open(item, 'r') as fh:
+                    line = fh.readline()
+                    print('line', line)
+                    debug_re = \
+                        re.compile('debug\s*=\s*([Tt]rue|[Ff]alse|[01])')
+                    while line:
+                        match = debug_re.search(line)
+                        if match:
+                            debug = match.group(1)
+                            break
+                        line = fh.readline()
+        if debug:
+            # our sys.exc_info queue should still be pointing at our ImportE
+            traceback.print_exc()
+
+    def __logger(self):
+        try:
+            self._logger = \
+                logging.getLogger('%s:%s' %
+                                  (self.default_project, self.default_name))
+            fh = \
+                logging.FileHandler("/var/log/%s/%s.log" %
+                                    (self.default_project, self.default_name))
+            fh.setLevel(logging.DEBUG)
+            self._logger.addHandler(fh)
+        except IOError:
+            self._logger = None
+
+    def __log_error(self):
+        if self._logger:
+            self._logger.fatal(self.message)
+        else:
+            syslog.syslog(syslog.LOG_CRIT, self.message)
+
+    def __set_message(self, args, kargs):
+        details = ', '.join(map(lambda x: (str(x)), args))
```

Comment:
I believe the following Right Hand Side of this assignment is easier to read, and gives you the same result:

`', '.join(map(str, t))`

If this doesn't handle a case, I'd be interested to learn that!